plutus-metatheory/src/Type/RenamingSubstitution.lagda.md

```diff
@@ -25,20 +25,35 @@ open _⊢⋆_
 
 ## Type renaming
 
-A type renaming is a mapping of type variables to type variables.
+Types can contain functions and as such are subject to a nontrivial equality relation. To
+explain the computation equation (the β-rule) we need to define substitution for a single
+type variable in a type. A type renaming is a function from type variables in one context to
+type variables in another. We only need the ability to
```

Comment:
We don't use Conor's [Type-Preserving Renaming and Substitution](http://strictlypositive.org/ren-sub.pdf) trick to express both renaming and substitution using a single general thing, right? Would that trick complicate things for us or do we not use it for a different reason?

Reply:
Right. That could save us some work, see later work by Allais et al which generalises renaming, substitution and evaluation. Note also you want to generalise the proofs not just the programs.  I didn't try this the first time around as I was more confident with ren+sub+evaluation defined and reasoned about separately. This has never got to the top of the list of priorities so far.

Reply:
Hereditary substitutions are also worth considering at some point.